src/server/frontend_xwayland/xwayland_server.cpp

```diff
@@ -431,8 +473,14 @@ mf::XWaylandServer::SocketFd::~SocketFd()
     unlink(path);
     snprintf(path, sizeof path, x11_socket_fmt, xdisplay);
     unlink(path);
-    close(abstract_socket_fd);
-    close(socket_fd);
+    if (abstract_socket_fd >= 0)
+    {
+        close(abstract_socket_fd);
+    }
+    if (socket_fd >= 0)
+    {
+        close(socket_fd);
+    }
```

Comment:
Should `abstract_socket_fd` and `socket_fd` be `mir::Fd`s?

Reply:
Obsoleted